libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts

```diff
@@ -136,4 +132,17 @@ export class ChoiceInput extends InputDialog {
         return `ChoiceInput[${ this.prompt && this.prompt.toString() }]`;
     }
 
+    private determineCulture(dc: DialogContext, opt: FindChoicesOptions = null): string {
+        const optLocale: string = opt && opt.locale ? opt.locale : null;
+        let culture: string = PromptCultureModels.mapToNearestLanguage(
+            dc.context.activity.locale ||
+            optLocale ||
+            (this.defaultLocale && this.defaultLocale.getValue(dc.state)));
+
+        if (!culture || !ChoiceInput.defaultChoiceOptions.hasOwnProperty(culture)) {
```

Comment:
It's a lot easier to read these boolean expressions if they are combined with `&&` and only undergo a single negation.

```typescript
if (!(culture && ChoiceInput.defaultChoiceOptions.hasOwnProperty(culure))) {
  ...
}
```
I see this logic in a few places so please update those as well.